bofire/data_models/strategies/fractional_factorial.py

```diff
@@ -0,0 +1,119 @@
+import re
+import string
+from itertools import combinations
+from typing import Annotated, Literal, Optional, Type
+
+from pydantic import Field, model_validator
+
+from bofire.data_models.constraints.api import Constraint
+from bofire.data_models.features.api import ContinuousInput, ContinuousOutput, Feature
+from bofire.data_models.strategies.strategy import Strategy
+
+
+class FractionalFactorialStrategy(Strategy):
+    type: Literal["FractionalFactorialStrategy"] = "FractionalFactorialStrategy"
+    n_repetitions: Annotated[int, Field(description="Number of repetitions", ge=0)] = 1
+    n_center: Annotated[int, Field(description="Number of center points", ge=0)] = 1
+    generator: Annotated[str, Field(description="Generator for the design.")] = ""
+    n_generators: Annotated[
+        int, Field(description="Number of reducing factors", ge=0)
+    ] = 0
+
+    @classmethod
+    def is_constraint_implemented(cls, my_type: Type[Constraint]) -> bool:
+        return False
+
+    @classmethod
+    def is_feature_implemented(cls, my_type: Type[Feature]) -> bool:
+        return my_type in [ContinuousOutput, ContinuousInput]
+
+    @model_validator(mode="after")
+    def validate(self):
+        if len(self.generator) > 0:
+            self.validate_generator(len(self.domain.inputs), self.generator)
+        else:
+            self.get_generator(
+                n_factors=len(self.domain.inputs), n_generators=self.n_generators
+            )
+        return self
+
+    @staticmethod
+    def validate_generator(n_factors: int, generator: str) -> str:
+        if len(generator.split(" ")) != n_factors:
+            raise ValueError("Generator does not match the number of factors.")
+        # clean it and transform it into a list
+        generators = [item for item in re.split(r"\-|\s|\+", generator) if item]
```

Comment:
Why do you split twice and differently? One time to just to check the length? And then with the regex again?

Reply:
Yes, once for the length and to remove "Vorzeichen". THe second one is stolen directly from pyDOE3.